index.js

```diff
@@ -96,89 +99,81 @@ function myMap(array, callback) {
   }
   return result
 }
-
 // ***** Callbacks - Question 1 *****
-
 // *** Uncomment the lines below to test
-// console.log("%cCallbacks - Question 1", "color: red")
-
-// console.log(myMap([1,2,3,4,5], triple)) 
+console.log("%cCallbacks - Question 1", "color: red")
+function triple(num) {
+  return num * 3
+}
+console.log(myMap([1,2,3,4,5], triple)) 
 // // => [3,6,9,12,15]
-
-// console.log(myMap([2,4,6,8], triple))   
+console.log(myMap([2,4,6,8], triple))   
 // // => [6,12,18,24]
-// console.log("%c----------", "color: red") 
-
-
+console.log("%c----------", "color: red") 
 // ***** Callbacks - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cCallbacks - Question 2", "color: red")
-
-// console.log(myMap(["Raffy", "Chase"], greet)) 
+console.log("%cCallbacks - Question 2", "color: red")
+function greet(name) {
+  return `Hi, ${name}`
+}
+console.log(myMap(["Raffy", "Chase"], greet)) 
 // // => ["Hi, Raffy!", "Hi, Chase!"]
-// console.log("%c----------", "color: red") 
-
-
-
+console.log("%c----------", "color: red") 
 // ***** Scope & Closures *****
-
 function takeANumber(line, name) {
   line.push(name)
-
   return `Welcome, ${name}. You are number ${line.length} in line.`
 }
-
 function nowServing(line) {
   if (!line.length) {
     return "There is nobody waiting to be served!"
   }
   return `Currently serving ${line.shift()}.`
 }
-
 const line = []
-
 // ***** Scope & Closures - Question 1 *****
-
-
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
-
-// console.log(takeATicketNumber(line))
+console.log("%cScope & Closures - Question 1", "color: red")
+let ticketNumber = 0
+function takeATicketNumber(line) {
+ticketNumber += 1
+line.push(ticketNumber)
+return `Welcome. You are ticket number ${ticketNumber}`
+}
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
-
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
-
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
-
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
-
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
-
-// console.log("%c----------", "color: red") 
-
+console.log("%c----------", "color: red") 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
-
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
+console.log("%cScope & Closures - Question 2", "color: red")
+// I ran out of time :(  Wanted to make sure I could submit! Also not at all confident on how to approach this yet. 
```

Comment:
No worries! You will basically just encapsulate all the logic from the previous question into one function.

```
function ticketNumberGeneratorFunc() {
  let number = 0;
  return (function (line) {
    number += 1;
    line.push(number);
    return `Welcome. You are ticket number ${number}`;
  })
}
```